swim/memberlist.go

```diff
@@ -250,11 +277,47 @@ func (m *memberlist) MakeFaulty(address string, incarnation int64) []Change {
 	return m.MakeChange(address, incarnation, Faulty)
 }
 
+// TODO: MakeLeave is only during testing. Need to figure out if we want to keep
+// it around or rewrite the tests.
 func (m *memberlist) MakeLeave(address string, incarnation int64) []Change {
+	// for backwards compatibility in tests we allow to update the local member
+	// in place when the gossip is about the local node and an incarnation number
+	// that is expected to have accept the update.
+	if address == m.local.Address && m.local.Incarnation <= incarnation {
+		m.local.Status = Leave
```

Comment:
Good note about the checksum being out of date after calling this. However calling `SetLocalStatus` also reincarnates the node by design which had a side effect on the tests where this function is being used. To be more clear, this function is _ONLY_ being used in tests at the moment.

I'm not happy that we still have this function in the membership anyway. I will have a second look in what it takes to remove this function.